lua/lazyvim/plugins/extras/lang/r.lua

```diff
@@ -47,6 +47,22 @@ return {
       require("r.pdf.generic").open = vim.ui.open
     end,
   },
+  {
+    "saghen/blink.cmp",
+    optional = true,
+    dependencies = {
+      "R-nvim/cmp-r",
+    },
+    opts = function(_, opts)
+      table.insert(opts.sources.compat or {}, "cmp_r")
+
+      opts.sources.providers["cmp-r"] = vim.tbl_deep_extend(
+        "force",
+        opts.sources.providers["cmp-r"] or {},
+        { name = "cmp_r", module = "blink.compat.source" }
+      )
+    end,
+  },
```

Comment:
Is this really needed? I think that is what is done when you just define the source name in `opts.sources.compat` either way.

Reply:
furthermore, `opts.sources.compat` is in list_extend, so you can just specify the option instead of a function.